partiql-logical/src/lib.rs

```diff
@@ -1,14 +1,58 @@
 use partiql_value::{BindingsName, Value};
-use petgraph::prelude::StableGraph;
-use petgraph::Directed;
 use std::collections::HashMap;
 
+#[derive(Debug, Hash)]
+pub struct OpId(usize);
+
+impl OpId {
+    pub fn index(&self) -> usize {
+        self.0
+    }
+}
+
+impl PartialEq<Self> for OpId {
+    fn eq(&self, other: &Self) -> bool {
+        self.index() == other.index()
+    }
+}
+
+impl Eq for OpId {}
+
 #[derive(Debug)]
-pub struct LogicalPlan(pub StableGraph<BindingsExpr, (), Directed>);
+pub struct LogicalPlan<'a, T> {
+    nodes: Vec<T>,
+    edges: Vec<(&'a OpId, &'a OpId)>,
+    node_count: usize,
```

Comment:
is this not redundant with `nodes.len()` ?

Reply:
e.g. 


```rust

pub fn node_count(&self) -> usize {self.nodes.len()}

```

Reply:
It is actually redundant, will change.